# Results

In [13]:
from pathlib import Path
import pandas as pd
from src.helpers.metrics import  build_metrics_table

from experiment_config import DegModel,DATA_NAME,ESTIMATION_NAME,PFNET_NAME,N_REP,PRED_STAT

DATA_NAME

'DS01'

In [14]:
#PFNET_NAME = 'net256x256x128x64x32leaky0.05_riskSL0.5'

estimation_dir = Path('experiments')/DATA_NAME/ESTIMATION_NAME
pred_dir = estimation_dir/DegModel.name()/'pf_performs'/PFNET_NAME/f"pred_{PRED_STAT}"

## Load Predictions

In [15]:
df_preds = pd.read_csv(pred_dir / f"rul_test_predictions_rep{N_REP}.csv")

## Compute mean prediction

In [16]:
df_preds = df_preds.groupby(["unit", "time"], as_index=False).mean()

## Full-life Predictions

In [17]:
metrics_df = build_metrics_table(df_preds)
metrics_df

,unit,RMSE,Score,PICP,PINAW
0,7,3.646116,28.785717,1.0,0.409617
1,8,8.327722,71.713665,1.0,0.409143
2,9,8.008107,94.859509,1.0,0.505898
3,10,3.664739,21.250477,1.0,0.481599


In [18]:
metrics_df.drop(columns=["unit"]).mean()

RMSE      5.911671
Score    54.152342
PICP      1.000000
PINAW     0.451564
dtype: float64

## Benchmark Predictions

In [22]:
df_benchmark = df_preds[df_preds["true_rul"] <= 65]

In [23]:
metrics_df = build_metrics_table(df_benchmark)
metrics_df

,unit,RMSE,Score,PICP,PINAW
0,7,3.914234,23.573703,1.0,0.397689
1,8,6.855733,39.368203,1.0,0.398643
2,9,7.714574,73.496508,1.0,0.533493
3,10,3.472870,15.473980,1.0,0.482836


In [24]:
metrics_df.drop(columns=["unit"]).mean()

RMSE      5.489353
Score    37.978099
PICP      1.000000
PINAW     0.453165
dtype: float64

## Degradation Predictions

In [19]:
hi_df = pd.read_csv(estimation_dir/'data_test.csv')
test_units = hi_df['unit'].astype(int).unique().tolist()
onsets = {unit: hi_df[(hi_df['unit']==unit) & (hi_df['hs']==0)]['cycle'].values[0] for unit in test_units}
df_degpreds = df_preds[
    df_preds["time"] > df_preds["unit"].map(onsets)
]

In [20]:
metrics_df = build_metrics_table(df_degpreds)
metrics_df

,unit,RMSE,Score,PICP,PINAW
0,7,3.333320,17.116974,1.0,0.393012
1,8,6.506782,34.795718,1.0,0.393434
2,9,5.976493,34.393031,1.0,0.574210
3,10,3.487183,14.780344,1.0,0.480708


In [21]:
metrics_df.drop(columns=["unit"]).mean()

RMSE      4.825945
Score    25.271517
PICP      1.000000
PINAW     0.460341
dtype: float64

| DS01	 | full      | RUL65     | deg       |
|--------|-----------|-----------|-----------|
| RMSE   | 5.068149  | 4.803219  | 4.711211  |
| Score  | 52.206928 | 37.873358 | 28.425670 |
| PICP   | 0.953896  | 0.943182  | 0.920168  |
| PINAW  | 0.312042  | 0.338091  | 0.337272  |

| DS01  | full      | RUL65     | deg       |
|-------|-----------|-----------|-----------|
| RMSE  | 5.911671  | 5.489353  | 4.825945  |
| Score | 54.152342 | 37.978099 | 25.271517 |
| PICP  | 1.000000  | 1.000000  | 1.000000  |
| PINAW | 0.451564  | 0.453165  | 0.460341  |

| DS03	 | full      | RUL65     | deg       |
|--------|-----------|-----------|-----------|
| RMSE   | 7.654103  | 7.383116  | 4.767812  |
| Score  | 71.544372 | 57.373969 | 22.033015 |
| PICP   | 0.926416  | 0.909519  | 0.913396  |
| PINAW  | 0.394850  | 0.398448  | 0.403939  |

| DS05   | full      | RUL65     | deg       |
|--------|-----------|-----------|-----------|
| RMSE   | 6.065781  | 5.153894  | 3.125471  |
| Score  | 45.452652 | 28.057284 | 13.593670 |
| PICP   | 0.988020  | 0.984848  | 0.980692  |
| PINAW  | 0.397921  | 0.422600  | 0.448572  |

| DS06   | full      | RUL65     | deg	     |
|--------|-----------|-----------|-----------|
| RMSE   | 6.697169  | 5.585590  | 4.804076  |
| Score  | 71.535756 | 39.490416 | 27.586838 |
| PICP   | 0.877232  | 0.920455  | 0.940313  |
| PINAW  | 0.324644  | 0.374098  | 0.430793  |